### DISCUSSION TITLE
Rendered code bringin JSON instead of HTML

### DISCUSSION THREAD

**Discussion by augustomallmann** [flareact/flareact#228] [Q&A]
Hello again,

 I noticed that the rendered HTML is not actually HTML, but it is JSON inside the body. Why is it happening?

URL to check the source code: https://umdu8r4e3w.map.azionedge.net/en/blog/caching
For example, this is the code from a dynamic route:

```
import type {
    FeaturedContentType,
    HubPageContent
} from '../../../src/types/hub';

import { CategoriesTab } from '../../../src/components/CategoriesTab';
import { ContentCardsGrid } from '../../../src/components/ContentCardsGrid';
import { CtaWithForm } from '../../../src/components/Cta';
import { FeaturedContent } from '../../../src/components/FeaturedContent';
import { Newsletter } from '../../../src/components/Newsletter';
import { Seo } from '../../../src/components/Seo';
import { getCategoryContent } from '../../../src/helpers/getCategoryContent';
import { getFeaturedContent } from '../../../src/helpers/getFeaturedContent';
import { useRouter } from 'flareact/router';

type HubPageProps = {
    featuredContent: FeaturedContentType[];
    content: HubPageContent;
};

export const Category = ({
    content,
    featuredContent
}: HubPageProps): JSX.Element => {
    const router = useRouter();
    const lang = router.query.lang;
    const { paginationOf } = content;

    return (
        <>
            <Seo
                title={paginationOf}
                description={paginationOf}
                openGraph={{
                    url: `https://www.azion.com/${lang}/blog/${paginationOf}`,
                    title: paginationOf,
                    description: 'description',
                    images: [{ url: 'https://www.azion.com/featured-image' }],
                    site_name: 'Azion Technologies'
                }}
            />
            <FeaturedContent content={featuredContent} />
            <Newsletter />
            <CategoriesTab currentTab={content.paginationOf} />
            <ContentCardsGrid content={content} />
            <CtaWithForm />
        </>
    );
};

export async function getEdgeProps({ params }) {
    const { lang, category } = params;
    const formatSlug = category.replace('/', '');
    const featuredContent = await getFeaturedContent(lang);
    const content = await getCategoryContent(lang, formatSlug);

    return {
        props: {
            content,
            featuredContent
        }
    };
}

export default Category;

```

**Comment by traffisco** [comment]
I would start in `_app.js` and `_document.js`. You somehow bypassing the renderer.

**Reply by augustomallmann** [reply]
I've been trying to find out why its not rendering by disabling codes and imports. What I know so far is that it only happens when I have a `getEdgeProps`, if I remove it, the HTML renders correctly. Do you have any suggestion on where should I continue this investigationg? I'm only fetching JSON code withing the `getEdgeProps` function.

**Reply by traffisco** [reply]
Its best if you can link to a public repo with minimal reproduction.
Try deleting the `out `folder and deleting the KV associated to project in CF dashboard.

**Comment by augustomallmann** [comment]
@traffisco hey, I'm back. Now I can share a public repo with you.
I deleted the `kv` and it didn't change anything. The out folder is deleted every build. 

What I notice now is that it doesn't render the HTML only when I use the data returned by the `getEdgeProps` function. Am I using it wrong ? 


**Edit:** I just solved the issue, so I deleted the links to the repository. What was causing problems with the render was the lib `framer motion`. Not sure why it was breaking everything, but after removing it, the app started to render in html.

Thanks so much for the help!!
